coffee_lib/src/plugin.rs

```diff
@@ -18,21 +22,34 @@ pub enum PluginLang {
 }
 
 impl PluginLang {
-    pub async fn default_install(&self, path: &str, name: &str) -> Result<String, CoffeeError> {
+    pub async fn default_install(&self, path: &str, name: &str, verbose: &bool) -> Result<String, CoffeeError> {
         match self {
             PluginLang::Python => {
                 /* 1. RUN PIP install or poetry install
                  * 2. return the path of the main file */
                 let req_file = format!("{path}/requirements.txt");
                 let main_file = format!("{path}/{name}.py");
                 // FIXME: enable the verbose command
-                let mut child = Command::new("pip")
+                if !verbose == false {
+                    // this is correct
+                    let mut child = Command::new("pip")
+                    .arg("install")
+                    .arg("-r")
+                    .arg(req_file.as_str())
+                    .spawn()
+                    .expect("not possible run the command");
+                    let _ = child.wait().await?;
+                } else {
+                    let mut child = Command::new("pip")
                     .arg("install")
+                    .arg("-q")
                     .arg("-r")
                     .arg(req_file.as_str())
                     .spawn()
                     .expect("not possible run the command");
-                let _ = child.wait().await?;
+                    let _ = child.wait().await?;
+                }
+                
```

Comment:
This works but now you are duplicating code, and also you do not prevent tokio to write to the std in this way.

The solution that I was in mind is something like that, when verbose is specified you do not use `spawn` but use output https://docs.rs/tokio/latest/tokio/process/struct.Command.html#method.output

```diff
+++ b/coffee_lib/src/plugin.rs
@@ -25,14 +25,17 @@ impl PluginLang {
                  * 2. return the path of the main file */
                 let req_file = format!("{path}/requirements.txt");
                 let main_file = format!("{path}/{name}.py");
-                // FIXME: enable the verbose command
-                let mut child = Command::new("pip")
-                    .arg("install")
-                    .arg("-r")
-                    .arg(req_file.as_str())
-                    .spawn()
-                    .expect("not possible run the command");
-                let _ = child.wait().await?;
+                let mut cmd = Command::new("pip");
+                cmd.arg("install").arg("-r").arg(&req_file.clone());
+                if verbose {
+                    let _ = cmd
+                        .spawn()
+                        .expect("Unable to run the command")
+                        .wait()
+                        .await?;
+                } else {
+                    let _ = cmd.output().await?;
+                };
                 Ok(main_file)
             }
```

In this way also future debug is more simple

What do you think?

Reply:
Hmm I get it. After reading the docs, the output function would be better to debug if any errors occur while installing the plugins, whereas using the flag `-q` won't offer any other utilities for debugging.